CoreInteractiveGraph/test/unit/src/au/gov/asd/tac/constellation/graph/interaction/plugins/io/AutosaveGraphPluginNGTest.java

```diff
@@ -0,0 +1,185 @@
+            if (tAttrId == Graph.NOT_FOUND) {
+                fail();
+            }
+
+            vxId1 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId1, 1.0f);
+            wg.setFloatValue(attrY, vxId1, 1.0f);
+            wg.setBooleanValue(vAttrId, vxId1, false);
+          //  wg.setStringValue(vAttrId, vxId1, "foo<Bar>");
+            vxId2 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId2, 5.0f);
+            wg.setFloatValue(attrY, vxId2, 1.0f);
+            wg.setBooleanValue(vAttrId, vxId2, true);
+            vxId3 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId3, 1.0f);
+            wg.setFloatValue(attrY, vxId3, 5.0f);
+            wg.setBooleanValue(vAttrId, vxId3, false);
+            vxId4 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId4, 5.0f);
+            wg.setFloatValue(attrY, vxId4, 5.0f);
+            wg.setBooleanValue(vAttrId, vxId4, false);
+            vxId5 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId5, 10.0f);
+            wg.setFloatValue(attrY, vxId5, 10.0f);
+            wg.setBooleanValue(vAttrId, vxId5, true);
+            vxId6 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId6, 15.0f);
+            wg.setFloatValue(attrY, vxId6, 15.0f);
+            wg.setBooleanValue(vAttrId, vxId6, false);
+            vxId7 = wg.addVertex();
+            wg.setFloatValue(attrX, vxId7, 100.0f);
+            wg.setFloatValue(attrY, vxId7, 100.0f);
+            wg.setBooleanValue(vAttrId, vxId7, true);
+
+            txId1 = wg.addTransaction(vxId1, vxId2, false);
+            txId2 = wg.addTransaction(vxId1, vxId3, false);
+            txId3 = wg.addTransaction(vxId2, vxId4, true);
+            txId4 = wg.addTransaction(vxId4, vxId2, true);
+            txId5 = wg.addTransaction(vxId5, vxId6, false);
+        } finally {
+            wg.commit();
+        }
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+        //graph = null;
+    }
+
+    /**
+     * Test of execute method, of class AutosaveGraphPlugin.
+     * @throws java.lang.Exception
+     */
+    @Test
+    public void testExecute() throws Exception {
+        final File saveDir = AutosaveUtilities.getAutosaveDir();
+        final File saveFile = new File(saveDir, graph.getId() + GraphDataObject.FILE_EXTENSION);
+        
+        // check the autosave file doesn't exist before running the plugin
+        assertEquals(saveFile.exists(), false);
+        
+        TopComponent tc = new TopComponent();
+        tc.setName("TestName");
+        final GraphDataObject gdo = GraphObjectUtilities.createMemoryDataObject("graph", true);
+        final GraphNode graphNode = new GraphNode(graph, gdo, tc, null);
+        AutosaveGraphPlugin instance = new AutosaveGraphPlugin();
+        PluginParameters parameters = instance.createParameters();
+        PluginExecution.withPlugin(instance).withParameters(parameters).executeNow(graph);
```

Comment:
Since the autosave graph plugin doesn't have any parameters, you can probably omit the creation of the parameters and the `.withParameters` call of the plugin execution